### PR TITLE
feat: add Icon

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

We are using here and there `<i className="fa fa-NAME"/>` for icons.
This create a dependency over font awesome and that is not validated at the moment.

**What is the new behavior?**

You can use <Icon name="NAME" /> and if the name starts with 'fa-' it will render a font awesome icon, if the NAME starts with icon- it will render a custom font icon and if it is **registred** it will use SVG

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Once this PR will be merge I will update all existings components

### CODE REVIEW

File: src/Icon/Icon.component.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * SVG implementation is inspired by
+ * http://svgicons.sparkk.fr/
+ * @param {object} props react props
+ * @example
+<Icon name="fa-bars"></Icon>
+ */
+function Icon(props) {
+	const id = props.name;
+	if (id.startsWith('fa-')) {
+		const className = `fa ${id}`;
```
Comment (talend-glorieux) [inline src/Icon/Icon.component.js]: Is that declaration needed?

File: src/Icon/Icon.component.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+	if (registry[id]) {
+		const classname = classnames(theme['svg-icon'], 'tc-svg-icon');
+		return (
+			<svg className={classname} viewBox="0 0 20 20">
+				{getSVG(id)}
+			</svg>
+		);
+	}
+	invariant(`icon not found ${id}`);
```
Comment (talend-glorieux) [inline src/Icon/Icon.component.js]: Isn't invariant supposed to use with two parameters `invariant(condition, message)`?

File: src/Icon/Icon.component.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	if (id.startsWith('fa fa-')) {
+		return (<i className={id} />);
+	}
+	if (id.startsWith('icon-')) {
+		return (<i className={id} />);
+	}
+	if (registry[id]) {
+		const classname = classnames(theme['svg-icon'], 'tc-svg-icon');
+		return (
+			<svg className={classname} viewBox="0 0 20 20">
```
Comment (acateland) [inline src/Icon/Icon.component.js]: the viewbox should be defined by the icon not by the svg element displaying the icon

File: src/Icon/Icon.component.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * SVG implementation is inspired by
+ * http://svgicons.sparkk.fr/
+ * @param {object} props react props
+ * @example
+<Icon name="fa-bars"></Icon>
+ */
+function Icon(props) {
+	const id = props.name;
+	if (id.startsWith('fa-')) {
+		return (<i className={`fa ${id}`} />);
```
Comment (frassinier) [inline src/Icon/Icon.component.js]: Must deal with a11y with `aria-hidden="true"` and `title` attributes. [More info](http://fontawesome.io/accessibility/)

File: src/Icon/Icon.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
+$tc-icon-color: #4691F6 !default;
```
Comment (frassinier) [inline src/Icon/Icon.scss]: Could you please add SassDoc?

File: src/Icon/Icon.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+.svg-icon {
```
Comment (frassinier) [inline src/Icon/Icon.scss]: Nesting feature could be great here, DRY

```
.svg-icon {
	width: 1em;
	height: 1em;

	path,
	polygon,
	rect {
		fill: $tc-icon-color;
	}

	circle {
		stroke: $tc-icon-color;
		stroke-width: 1;
	}
}
```

File: stories/Icon.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import { Icon } from '../src/index';
+
+Icon.register('svg-test', (<path d="M10.219,1.688c-4.471,0-8.094,3.623-8.094,8.094s3.623,8.094,8.094,8.094s8.094-3.623,8.094-8.094S14.689,1.688,10.219,1.688 M10.219,17.022c-3.994,0-7.242-3.247-7.242-7.241c0-3.994,3.248-7.242,7.242-7.242c3.994,0,7.241,3.248,7.241,7.242C17.46,13.775,14.213,17.022,10.219,17.022 M15.099,7.03c-0.167-0.167-0.438-0.167-0.604,0.002L9.062,12.48l-2.269-2.277c-0.166-0.167-0.437-0.167-0.603,0c-0.166,0.166-0.168,0.437-0.002,0.603l2.573,2.578c0.079,0.08,0.188,0.125,0.3,0.125s0.222-0.045,0.303-0.125l5.736-5.751C15.268,7.466,15.265,7.196,15.099,7.03" />));
+
+storiesOf('Icon', module)
+	.addWithInfo('default', () => (
+		<div>
+			<h1>Icon</h1>
+			<h2>Definition</h2>
+			<p>The Icon component display an icon (sic).</p>
```
Comment (frassinier) [inline stories/Icon.js]: Typo 

The Icon component display**s** [...]

File: src/Icon/Icon.component.js (hunk 1 above)
Comment (jmfrancois) [inline src/Icon/Icon.component.js]: no, change done

File: src/Icon/Icon.component.js (hunk 2 above)
Comment (jmfrancois) [inline src/Icon/Icon.component.js]: oups, exact

File: src/Icon/Icon.component.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+ * http://svgicons.sparkk.fr/
+ * @param {object} props react props
+ * @example
+<Icon name="fa-bars"></Icon>
+ */
+function Icon(props) {
+	const id = props.name;
+	const accessibility = {
+		'aria-hidden': 'true',
+		title: props.title || null,
```
Comment (frassinier) [inline src/Icon/Icon.component.js]: if `props.title` is null is equivalent to `title:null || null` but I like the way you deal with a11y attributes :)

**Pull request merged**